### PR TITLE
Setup Firebase live-database

### PR DESCRIPTION
**Pull request by webber-jiang**

### CODE REVIEW

File: src/components/recipeList/RecipeList.tsx [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+} from '../../actions/index';
 import { RecipeConfig, RecipesApiConfig } from '../../actions/types';
 
 const RecipeList = (): JSX.Element => {
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    dispatch(getRecipeFromFavorites());
+  }, [dispatch]);
+
```
Comment (Ray2126) [inline src/components/recipeList/RecipeList.tsx:19]: Does this get the favourited recipes from the firebase database and put it into the Redux store?

File: src/util/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  databaseURL: 'https://diet-app-3c1a9.firebaseio.com',
+  projectId: process.env.PROJECT_ID,
+  storageBucket: process.env.STORAGE_BUCKET,
+  messagingSenderId: process.env.MESSAGING_SENDER_ID,
+  appId: process.env.APP_ID,
+  measurementId: process.env.MEASUREMENT_ID,
+};
+// Initialize Firebase
+firebase.initializeApp(firebaseConfig);
+// firebase.analytics();
```
Comment (Ray2126) [inline src/util/firebase.ts:17]: Remove these comments if you don't need them.

File: src/actions/index.ts [hunk 3]
```diff
@@ -37,10 +40,42 @@ export const getRecipe = (recipe: RecipeConfig): ActionConfig => {
...
-export const removeFromFavorites = (recipe: RecipeConfig): ActionConfig => {
-  return { type: REMOVE_RECIPE_FROM_FAVORITES, payload: recipe };
+export const removeFromFavorites = (recipe: RecipeConfig) => async (
+  dispatch: Dispatch
+): Promise<void> => {
+  dispatch({ type: REMOVE_RECIPE_FROM_FAVORITES, payload: recipe });
+  firebase
+    .database()
+    .ref('favorites')
+    .orderByChild('id')
```
Comment (Ray2126) [inline src/actions/index.ts:57]: Do you need to order the recipes here? Aren't you just looking for the one recipe to remove?

Comment (webber-jiang) [reply src/actions/index.ts:57]: I'll double check, I was just going through the Firebase Docs.

File: src/util/firebase.ts (hunk 2 above)
Comment (webber-jiang) [reply src/util/firebase.ts:17]: This will be needed later

File: src/components/recipeList/RecipeList.tsx (hunk 1 above)
Comment (webber-jiang) [reply src/components/recipeList/RecipeList.tsx:19]: Yes, I dispatch it to the reducer, have a look at the actions.

File: src/util/firebase.ts (hunk 2 above)
Comment (webber-jiang) [reply src/util/firebase.ts:17]: Actually, Im gonna leave these in here, its just a config

File: src/actions/index.ts (hunk 3 above)
Comment (webber-jiang) [reply src/actions/index.ts:57]: If I remove it, it doesn't work, must be a Firebase thing.

Comment (Ray2126) [reply src/actions/index.ts:57]: Ok it's fine then.

**Review by Ray2126 — APPROVED** [review]

**Pull request merged**